src/server/game/AI/SmartScripts/SmartScript.cpp

```diff
@@ -847,23 +855,28 @@ void SmartScript::ProcessAction(SmartScriptHolder& e, Unit* unit, uint32 var0, u
         }
         case SMART_ACTION_CALL_GROUPEVENTHAPPENS:
         {
-            if (!unit)
```

Comment:
Restore this check, you are dereferencing unit below, which could lead to a segmentation fault crash.

Reply:
I'll try it with it

Reply:
After that, task 12687 cannot be completed

Reply:
I don't understand, but don't remove that check or we will have a crash later. If there is an issue with something else, then is an issue with the script itself and not with the SAI system

Reply:
ok